common/types.go

```diff
@@ -301,6 +301,30 @@ func SetField(obj map[string]interface{}, k string, v interface{}) bool {
 	return true
 }
 
+// DelField detete a value in a tree based on dot key
+func DelField(obj map[string]interface{}, k string) interface{} {
+	components := strings.Split(k, ".")
+	o, ok := obj[components[0]]
+	if !ok {
+		return obj
+	}
+	if len(components) == 1 {
+		delete(obj, k)
+		return obj
+	}
+
+	object, ok := o.(map[string]interface{})
+	if !ok {
+		return obj
+	}
+	r := DelField(object, strings.SplitN(k, ".", 2)[1])
```

Comment:
sorry didn't see that before but you need to check that r is map[string]interface{} before using len on it as it can be something else in case of line 318